stdlib/public/core/RuntimeFunctionCounters.swift

```diff
@@ -0,0 +1,506 @@
+    // Bail if this object was seen already.
+    if visitedItems[theId] != nil {
+      return
+    }
+    // Remember that this object was seen already.
+    let identifier = visitedItems.count
+    visitedItems[theId] = identifier
+  }
+
+  // If it is a reference, add it to the result.
+  if let ref = ref {
+    references.append(ref)
+  }
+
+  // Recursively visit the children of the current value.
+  var currentIndex = mirror.children.startIndex
+  for _ in 0..<count {
+    let (_, child) = mirror.children[currentIndex]
+    mirror.children.formIndex(after: &currentIndex)
+    _collectAllReferencesInsideObjectImpl(
+      child,
+      references: &references,
+      visitedItems: &visitedItems)
+  }
+}
+
+/// This is a namespace for runtime functions related to management
+/// of runtime function counters.
+public // @testable
+struct _RuntimeFunctionCounters {
+  public typealias RuntimeFunctionCountersUpdateHandler =
+   @convention(c) (_ object: UnsafeRawPointer, _ functionId: Int64) -> Void
+
+  public static let runtimeFunctionNames =
+    getRuntimeFunctionNames()
+  public static let runtimeFunctionCountersOffsets =
+    _RuntimeFunctionCounters.getRuntimeFunctionCountersOffsets()
+  public static let numRuntimeFunctionCounters =
+    _RuntimeFunctionCounters.getNumRuntimeFunctionCounters()
+  public static let runtimtFunctionNameToIndex: [String : Int] =
+    getRuntimeFunctionNameToIndex()
+
+  /// Get the names of all runtime functions whose calls are being
+  /// tracked.
+  @_silgen_name("getRuntimeFunctionNames")
+  static public func _getRuntimeFunctionNames() ->
+    UnsafePointer<UnsafePointer<CChar>>
+
+  static public func getRuntimeFunctionNames() -> [String] {
+    let runtimeFunctionNames = _RuntimeFunctionCounters._getRuntimeFunctionNames()
+    let numRuntimeFunctionCounters =
+      _RuntimeFunctionCounters.getNumRuntimeFunctionCounters()
+    var runtimtFunctionNames : [String] = []
+    for index in 0..<numRuntimeFunctionCounters {
+      let name = String(cString:runtimeFunctionNames[index])
+      runtimtFunctionNames.append(name)
+    }
+    return runtimtFunctionNames
+  }
+
+  /// Get the offsets of the collected runtime function counters inside
+  /// the state.
+  @_silgen_name("getRuntimeFunctionCountersOffsets")
+  static public func getRuntimeFunctionCountersOffsets() ->
+    UnsafePointer<UInt16>
+
+  /// Get the number of different runtime functions whose calls are being
+  /// tracked.
+  @_silgen_name("getNumRuntimeFunctionCounters")
+  static public func getNumRuntimeFunctionCounters() -> Int
+
+  /// Dump all per-object runtime function counters.
+  @_silgen_name("dumpObjectsRuntimeFunctionPointers")
+  static public func dumpObjectsRuntimeFunctionPointers()
+
+  @_silgen_name("setGlobalRuntimeFunctionCountersUpdateHandler")
+  static public func setGlobalRuntimeFunctionCountersUpdateHandler(
+    handler: RuntimeFunctionCountersUpdateHandler?) ->
+    RuntimeFunctionCountersUpdateHandler?
```

Comment:
I started to prefer the formatting where `->` is on the same line with the function result type. As in:

```swift
static public func setGlobalRuntimeFunctionCountersUpdateHandler(
  handler: RuntimeFunctionCountersUpdateHandler?
) -> RuntimeFunctionCountersUpdateHandler?
```